Documentation/design-docs/arm64-intrinsics.md

```diff
@@ -0,0 +1,217 @@
+## Arm64 Intrinsics
+
+This document is intended to document proposed design decision related to the introduction
+of Arm64 Intrinsics
+
+### Intrinsics in general
+
+Use of intrinsics in general is a CoreCLR design decision to allow low level platform
+specific optimizations.  
+
+At first glance, such a decision seems to violate the fundamental principles of .NET
+code running on any platform.  However, the intent is not for the vast majority of
+apps to use such optimizations.  The intended usage model is to allow library
+developers access to low level functions which enable optimization of key
+functions.  As such the use is expected to be limited, but performance critical.
+
+#### Intrinsic `IsSupported` granularity
+
+The existing implementation is using `IsSupported` to determine whether an existing
+namespace is supported on a given platform.
+
+The granularity is kept at a large scale to have large blocks of functionality controlled
+by a single `IsSupported`.
+
+#### Intrinsic granularity
+
+In general individual intrinsic will be chosen to be fine grained.  These will generally correspond to a single assembly instruction.
+
+#### Intrinsics & Crossgen
+
+For any intrinsic which may not be supported on all variants of a platform.  Crossgen
+Method compilation must be trapped, so that the JIT is forced to generate optimal platform dependent 
+code at runtime.
```

Comment:
That is an open problem with several solutions proposed so far. Probably there is no good reason to limit the choice in design docs.

Reply:
As design dos intended to guide implementation.  I need it in the doc to remind me to implement the feature